python/pyiceberg/table/partitioning.py

```diff
@@ -29,7 +29,7 @@
 from pyiceberg.transforms import Transform
 from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
 
-INITIAL_SPEC_ID = 0
+INITIAL_PARTITION_SPEC_ID = 1
```

Comment:
I think it's a good idea to start assigning spec IDs at 1 and reserve 0 for unpartitioned, but there's no requirement in the spec to do that. Plus, it would be a breaking change so we can't really fix it.

Reply:
Alright, moved everything back